File: leetcode/array/1.twoSum/1twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const memo = {};
    for (let i = 0; i < nums.length; i++) {
        const another = target - nums[i];
        if (memo[another] !== undefined) {
            return [memo[another], i];
        }
        memo[nums[i]] = i;
    }
    return [];
};

File: leetcode/array/1.twoSum/1twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        for (int i = 0; i < nums.size(); i++)
        {
            int t = target - nums[i];
            if (hash.count(t)) return vector<int>({hash[t], i});
            hash[nums[i]] = i;
        }
        return vector<int>();
    }
};